packages/fiori/src/DynamicSideContent.js

```diff
@@ -0,0 +1,546 @@
+		return metadata;
+	}
+
+	static get styles() {
+		return DynamicSideContentCss;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get template() {
+		return DynamicSideContentTemplate;
+	}
+
+	static async onDefine() {
+		DynamicSideContent.i18nBundle = await getI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	get classes() {
+		const gridPrefix = "ui5-dsc-span",
+			mcSpan = this._toggled ? this._scSpan : this._mcSpan,
+			scSpan = this._toggled ? this._mcSpan : this._scSpan,
+			classes = {
+				dynamicSideContentRoot: {
+					"ui5-dsc-root": true,
+				},
+				mainContent: {
+					"ui5-dsc-main": true,
+				},
+				sideContent: {
+					"ui5-dsc-side": true,
+				},
+			};
+
+		classes.mainContent[`${gridPrefix}-${mcSpan}`] = true;
+		classes.sideContent[`${gridPrefix}-${scSpan}`] = true;
+
+		return classes;
+	}
+
+	get styles() {
+		const isToggled = this.breakpoint === this.sizeS && this._toggled,
+			mcSpan = isToggled ? this._scSpan : this._mcSpan,
+			scSpan = isToggled ? this._mcSpan : this._scSpan,
+			contentHeight = this.breakpoint === this.sizeS && this.sideContentVisibility !== SideContentVisibility.AlwaysShow ? "100%" : "auto";
+
+		return {
+			dynamicSideContentRoot: {
+				"height": "100%",
+				"flex-wrap": this._mcSpan === "12" ? "wrap" : "nowrap",
+			},
+			mainContent: {
+				"height": mcSpan === this.span12 ? contentHeight : "100%",
+				"order": this.sideContentPosition === SideContentPosition.Begin ? 2 : 1,
+			},
+			sideContent: {
+				"height": scSpan === this.span12 ? contentHeight : "100%",
+				"order": this.sideContentPosition === SideContentPosition.Begin ? 1 : 2,
+			},
+		};
+	}
+
+	get accInfo() {
+		return {
+			"label": DynamicSideContent.i18nBundle.getText(DSC_SIDE_ARIA_LABEL),
+		};
+	}
+
+	get sizeS() {
+		return "S";
+	}
+
+	get sizeM() {
+		return "M";
+	}
+
+	get sizeL() {
+		return "L";
+	}
+
+	get sizeXL() {
+		return "XL";
+	}
+
+	get span0() {
+		return "0";
+	}
+
+	get span3() {
+		return "3";
+	}
+
+	get span4() {
+		return "4";
+	}
+
+	get span6() {
+		return "6";
+	}
+
+	get span8() {
+		return "8";
+	}
+
+	get span9() {
+		return "9";
+	}
+
+	get span12() {
+		return "12";
+	}
+
+	get spanFixed() {
+		return "fixed";
+	}
+
+	get containerWidth() {
+		return this.containerQuery ? this.getDomRef().clientWidth : window.innerWidth;
+	}
+
+	get breakpoint() {
+		const S_M_BREAKPOINT = 720,	// Breakpoint between S and M screen size
+			M_L_BREAKPOINT = 1024, // Breakpoint between M and L screen size
+			L_XL_BREAKPOINT = 1440; // Breakpoint between L and XL screen size
+		let size;
+
+		if (this.containerWidth <= S_M_BREAKPOINT) {
+			size = this.sizeS;
+		} else if (this.containerWidth > S_M_BREAKPOINT && this.containerWidth <= M_L_BREAKPOINT) {
+			size = this.sizeM;
+		} else if (this.containerWidth > M_L_BREAKPOINT && this.containerWidth <= L_XL_BREAKPOINT) {
+			size = this.sizeL;
+		} else {
+			size = this.sizeXL;
+		}
+
+		return size;
+	}
+
+	toggleContents() {
```

Comment:
If this is meant to be a public method you have to document it otherwise no would no that it exists.

Reply:
Done